charts/pulsar/templates/broker/broker-statefulset.yaml

```diff
@@ -219,6 +219,7 @@ spec:
           initialDelaySeconds: {{ .Values.broker.probe.liveness.initialDelaySeconds }}
           periodSeconds: {{ .Values.broker.probe.liveness.periodSeconds }}
           failureThreshold: {{ .Values.broker.probe.liveness.failureThreshold }}
+          timeoutSeconds: {{ .Values.broker.probe.liveness.timeoutSeconds }}
```

Comment:
why do we add this timeoutSeconds?

Reply:
In my test, I found broker will fail to start and loop in restarting for several times. I captured some logs, looks like relate to the function worker in the broker?

```
10:51:04.244 [AsyncHttpClient-53-1] WARN  org.apache.pulsar.client.admin.internal.BaseResource - [http://pulsar-broker-1.pulsar-broker.pulsar.svc.cluster.local:8080/admin/v2/persistent/public/functions/assignments] Failed to perform http put request: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: pulsar-broker-2.pulsar-broker.pulsar.svc.cluster.local
10:51:04.245 [main] ERROR org.apache.pulsar.functions.worker.PulsarWorkerService - Error Starting up in worker
org.apache.pulsar.client.admin.PulsarAdminException: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: pulsar-broker-2.pulsar-broker.pulsar.svc.cluster.local
	at org.apache.pulsar.client.admin.PulsarAdminException.wrap(PulsarAdminException.java:252) ~[io.streamnative-pulsar-client-admin-api-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.client.admin.internal.BaseResource.sync(BaseResource.java:302) ~[io.streamnative-pulsar-client-admin-original-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.client.admin.internal.TopicsImpl.createNonPartitionedTopic(TopicsImpl.java:340) ~[io.streamnative-pulsar-client-admin-original-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.client.admin.Topics.createNonPartitionedTopic(Topics.java:482) ~[io.streamnative-pulsar-client-admin-api-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.functions.worker.PulsarWorkerService.tryCreateNonPartitionedTopic(PulsarWorkerService.java:390) ~[io.streamnative-pulsar-functions-worker-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.functions.worker.PulsarWorkerService.start(PulsarWorkerService.java:441) ~[io.streamnative-pulsar-functions-worker-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.broker.PulsarService.startWorkerService(PulsarService.java:1624) ~[io.streamnative-pulsar-broker-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.broker.PulsarService.start(PulsarService.java:808) ~[io.streamnative-pulsar-broker-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.PulsarBrokerStarter$BrokerStarter.start(PulsarBrokerStarter.java:273) ~[io.streamnative-pulsar-broker-2.10.3.6.jar:2.10.3.6]
	at org.apache.pulsar.PulsarBrokerStarter.main(PulsarBrokerStarter.java:350) ~[io.streamnative-pulsar-broker-2.10.3.6.jar:2.10.3.6]
	Suppressed: org.apache.pulsar.client.admin.PulsarAdminException: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: pulsar-broker-2.pulsar-broker.pulsar.svc.cluster.local
		at org.apache.pulsar.client.admin.internal.BaseResource.getApiException(BaseResource.java:251) ~[io.streamnative-pulsar-client-admin-original-2.10.3.6.jar:2.10.3.6]
		at org.apache.pulsar.client.admin.internal.BaseResource$1.failed(BaseResource.java:134) ~[io.streamnative-pulsar-client-admin-original-2.10.3.6.jar:2.10.3.6]
		at org.glassfish.jersey.client.JerseyInvocation$1.failed(JerseyInvocation.java:882) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
		at org.glassfish.jersey.client.ClientRuntime.processFailure(ClientRuntime.java:247) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
		at org.glassfish.jersey.client.ClientRuntime.processFailure(ClientRuntime.java:242) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
		at org.glassfish.jersey.client.ClientRuntime.access$100(ClientRuntime.java:62) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
		at org.glassfish.jersey.client.ClientRuntime$2.lambda$failure$1(ClientRuntime.java:178) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
		at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
		at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
		at org.glassfish.jersey.internal.Errors.process(Errors.java:292) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
		at org.glassfish.jersey.internal.Errors.process(Errors.java:274) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
		at org.glassfish.jersey.internal.Errors.process(Errors.java:244) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
		at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:288) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
		at org.glassfish.jersey.client.ClientRuntime$2.failure(ClientRuntime.java:178) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
		at org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector.lambda$apply$1(AsyncHttpConnector.java:223) ~[io.streamnative-pulsar-client-admin-original-2.10.3.6.jar:2.10.3.6]
		at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
		at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
		at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
		at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) ~[?:?]
		at org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector.lambda$retryOperation$4(AsyncHttpConnector.java:282) ~[io.streamnative-pulsar-client-admin-original-2.10.3.6.jar:2.10.3.6]
		at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
		at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
		at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
		at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) ~[?:?]
		at org.asynchttpclient.netty.NettyResponseFuture.abort(NettyResponseFuture.java:273) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.request.NettyRequestSender.abort(NettyRequestSender.java:473) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.request.NettyRequestSender$1.onFailure(NettyRequestSender.java:325) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.SimpleFutureListener.operationComplete(SimpleFutureListener.java:26) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:185) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:35) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at org.asynchttpclient.netty.request.NettyRequestSender.sendRequestWithNewChannel(NettyRequestSender.java:301) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.request.NettyRequestSender.sendRequestWithCertainForceConnect(NettyRequestSender.java:142) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.request.NettyRequestSender.sendRequest(NettyRequestSender.java:113) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.request.NettyRequestSender.sendNextRequest(NettyRequestSender.java:548) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.handler.intercept.Redirect30xInterceptor.exitAfterHandlingRedirect(Redirect30xInterceptor.java:160) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.handler.intercept.Interceptors.exitAfterIntercept(Interceptors.java:98) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.handler.HttpHandler.handleHttpResponse(HttpHandler.java:78) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.handler.HttpHandler.handleRead(HttpHandler.java:140) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at org.asynchttpclient.netty.handler.AsyncHttpClientHandler.channelRead(AsyncHttpClientHandler.java:78) ~[org.asynchttpclient-async-http-client-2.12.1.jar:?]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[io.netty-netty-codec-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[io.netty-netty-codec-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[io.netty-netty-codec-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[io.netty-netty-transport-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[io.netty-netty-common-4.1.87.Final.jar:4.1.87.Final]
		at java.lang.Thread.run(Thread.java:829) ~[?:?]
	Caused by: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: pulsar-broker-2.pulsar-broker.pulsar.svc.cluster.local
		at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
		at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
		at java.util.concurrent.CompletableFuture$OrApply.tryFire(CompletableFuture.java:1503) ~[?:?]
		... 52 more
```

Reply:
Removed in https://github.com/streamnative/charts/pull/985/commits/85d06b27346c1651db1940ae7b15c99a3b234d85 I think the error is from the file permission for `connectorsDirectory`.